examples/layertree.js

```diff
@@ -100,7 +126,19 @@ app.LayertreeController.prototype.getLayer = function(node) {
  * @export
  */
 app.LayertreeController.prototype.onButtonClick = function(node, layer) {
-  window.alert(node['name'] + ', ' + layer.get('type'));
+  if (!angular.isDefined(node.metadataPromise_)) {
```

Comment:
This would work in the example since the url is the same for every layer. However, in a real application one will want to call a different service.